common/utilities/setQueryParameters.js

```diff
@@ -0,0 +1,22 @@
+import Router from 'next/router'
+import { isBrowser } from './isBrowser'
+import { convertStringToBoolean } from './convertStringToBoolean'
+
+export const setQueryParameters = query => {
+  if(!isBrowser()){
+    return
+  }
+
+  const previousQuery = Router.router.query
+  const filteredQuery = _.pickBy(query, query => !_.isNil(query) && query !== '')
+  const parsedPreviousQuery = _.mapValues(previousQuery, query => query === 'true' || query === 'false' ? convertStringToBoolean(query) : query)
```

Comment:
this code is a bit hard to understand and just doesn't feel right.

You need to compare two things are equal. And you've taken the query parameters where things are strings and converted them back to booleans.

How about we do it the other way around? Take the current query, and `stringifyQueryValues(` then you can just straight up compare the old values. You can use this function before you save them as well. 

Side note: mapping over all the values to change just some felt weird to me, would have just done the ones I knew were booleans.

Reply:
There's quite a lot of heavy 'doing' stuff in this function. So I think breaking it out into a nice function or two will increase readability alot.